src/Microsoft.Azure.SignalR.Serverless.Protocols/Internal/MemoryBufferWriter.cs

```diff
@@ -0,0 +1,342 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.SignalR.Serverless.Protocols
+{
+    internal sealed class MemoryBufferWriter : Stream, IBufferWriter<byte>
```

Comment:
looks like this is copied from https://github.com/dotnet/aspnetcore/blob/master/src/SignalR/common/Shared/MemoryBufferWriter.cs, keep it as it is and add comment about the reference?